src/Compilers/Core/Portable/SymbolDisplay/SymbolDisplayFormat.cs

```diff
@@ -753,6 +806,31 @@ public SymbolDisplayFormat RemoveLocalOptions(SymbolDisplayLocalOptions options)
             return this.WithLocalOptions(this.LocalOptions & ~options);
         }
 
+        /// <summary>
+        /// Creates a copy of the <see cref="SymbolDisplayFormat"/> but with replaced <see cref="ConstantValueOptions"/>.
+        /// </summary>
+        /// <param name="options">
+        /// An object representing how constant values will be displayed.
+        /// </param>
+        /// <returns>A duplicate of the <see cref="SymbolDisplayFormat"/>, with replaced <see cref="ConstantValueOptions"/>.</returns>
+        public SymbolDisplayFormat WithConstantValueOptions(SymbolDisplayConstantValueOptions options)
+        {
+            return new SymbolDisplayFormat(
+                this.CompilerInternalOptions,
+                this.GlobalNamespaceStyle,
+                this.TypeQualificationStyle,
+                this.GenericsOptions,
+                this.MemberOptions,
+                this.ParameterOptions,
+                this.DelegateStyle,
+                this.ExtensionMethodStyle,
+                this.PropertyStyle,
+                this.LocalOptions,
+                this.KindOptions,
+                this.MiscellaneousOptions,
+                options);
```

Comment:
wow.  :)  why isn't this type using the standard 'With' impl.  that would make this code into simply: `return this.With(constantVallueOptions: options)` and would make all the other methods so much shorter.